File: 1001-1500/1229_minAvailableDuration.py
class Solution:
    def minAvailableDuration(self, slots1: List[List[int]], slots2: List[List[int]], duration: int) -> List[int]:
        slots1.sort(key = lambda x: x[0])
        slots2.sort(key = lambda x: x[0])
        interval, i, j = [], 0, 0
        while i < len(slots1) and j < len(slots2):
            if slots1[i][1] <= slots2[j][0]:
                i += 1
            elif slots2[j][1] <= slots1[i][0]:
                j += 1
            else:
                start = max(slots1[i][0], slots2[j][0])
                end = min(slots1[i][1], slots2[j][1])
                if end - start >= duration:
                    interval = [start, start + duration]
                    break
                else:
                    if slots1[i][1] < slots2[j][1]:
                        i += 1
                    else:
                        j += 1
        return interval

File: 1001-1500/1229_minAvailableDuration.java
class Solution {
    public List<Integer> minAvailableDuration(int[][] slots1, int[][] slots2, int duration) {
        Arrays.sort(slots1, (a, b) -> Integer.compare(a[0], b[0]));
        Arrays.sort(slots2, (a, b) -> Integer.compare(a[0], b[0]));
        
        int i = 0;
        int j = 0;
        
        while (i < slots1.length && j < slots2.length) {
            int left = Math.max(slots1[i][0], slots2[j][0]);
            
            if (slots1[i][1] < slots2[j][1]) {
                int right = slots1[i][1];
                if (right - left >= duration) {
                    return List.of(left, left + duration);
                }
                i++;
            } else if (slots1[i][1] > slots2[j][1]) {
                int right = slots2[j][1];
                if (right - left >= duration) {
                    return List.of(left, left + duration);
                }
                j++;
            } else {
                int right = slots2[j][1];
                if (right - left >= duration) {
                    return List.of(left, left + duration);
                }
                i++;
                j++;
            }
        }
        
        return new ArrayList<>();
    }
}
